src/Integration/NewConnectedMode/UnbindCommand.cs

```diff
@@ -0,0 +1,88 @@
+/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Integration.Resources;
+
+namespace SonarLint.VisualStudio.Integration.NewConnectedMode
+{
+    internal class UnbindCommand
+    {
+        private readonly IHost host;
+        private readonly IConfigurationProvider configProvider;
+
+        public UnbindCommand(IHost host)
+        {
+            if (host == null)
+            {
+                throw new ArgumentNullException(nameof(host));
+            }
+            this.host = host;
+
+            this.configProvider = this.host.GetService<IConfigurationProvider>();
+            configProvider.AssertLocalServiceIsNotNull();
+        }
+
+        public bool CanExecute()
+        {
+            // Can only unbind if currently in new bound mode
+            return  !host.VisualStateManager.IsBusy &&
+                configProvider.GetConfiguration().Mode == SonarLintMode.Connected;
+        }
+
+        public void Execute()
+        {
+            Debug.Assert(this.CanExecute(), $"Should not be called UnbindCommand.Execute if CanExecute is false");
```

Comment:
$ is useless + I think the sentence has a problem and should either be:
_Should not be called if CanExecute is false_
or
_Should not have called UnbindCommand.Execute if CanExecute is false_